src/Symfony/Component/HttpClient/Response/AsyncResponse.php

```diff
@@ -0,0 +1,349 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Response;
+
+use Symfony\Component\HttpClient\Chunk\ErrorChunk;
+use Symfony\Component\HttpClient\Chunk\LastChunk;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Provides a single extension point to process a response's content stream.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class AsyncResponse implements ResponseInterface
+{
+    use CommonResponseTrait;
+
+    private $client;
+    private $response;
+    private $info = ['canceled' => false];
+    private $passthru;
+
+    /**
+     * @param callable(ChunkInterface, AsyncContext): ?\Iterator $passthru
```

Comment:
Should we enforce `\Iterator` or could `\Traversable` be enough ?

Reply:
we must enforce `Iterator` since we're using its methods directly.